src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/generator/flow/MultiTableServer42IngressRuleGenerator.java

```diff
@@ -87,10 +87,14 @@ public class MultiTableServer42IngressRuleGenerator extends Server42IngressRuleG
     public List<SpeakerData> generateCommands(Switch sw) {
         List<SpeakerData> result = new ArrayList<>();
         if (switchProperties == null || !switchProperties.isServer42FlowRtt()
-                || flow == null || flow.isOneSwitchFlow()) {
+                || (flow != null && flow.isOneSwitchFlow())) {
```

Comment:
The condition became complex and confusing, easy to make a typo, consider to extract it to a method with some descriptive name and a unit test.